### PR TITLE
Improve display of combobox windows

### PR DESCRIPTION
**Pull request by amousset**
* align window with the button
* avoid moving the current choice when dropdown windows opens
* align button to the left in preferences

combo2
combo3
combo1

### CODE REVIEW

File: data/themes/darktable.css [hunk 1]
```diff
@@ -870,9 +886,9 @@ menuitem check
...
 
-/* Header and footer combobox */
-#header-toolbar combobox,
-#footer-toolbar combobox
+/* Header and footer labels */
+#header-toolbar label,
+#footer-toolbar label
 {
   margin: 0 0.5em;
 }
```
Comment (Nilvus) [inline data/themes/darktable.css:894]: Do not remove combobox here, add it between toolbar and label for both header and footer line. That will avoid having that setting applied for some label we have outside a combobox.

File: data/themes/darktable.css [hunk 2]
```diff
@@ -1073,7 +1089,7 @@ dialog scrollbar
   margin: 0px; /* reset default dialog margin to all items. And needed to have an hover and selected effect on all width of categories in sidebar */
 }
 
-#preferences_notebook decoration  /* allow to resize preferences dialog window */
+#preferences_notebook dialog > decoration  /* allow to resize preferences dialog window */
```
Comment (Nilvus) [inline data/themes/darktable.css:1092]: A change really not useful as it works without it.

File: data/themes/darktable.css [hunk 3]
```diff
@@ -825,16 +841,16 @@ dialog combobox window
...
   background-color: @tooltip_bg_color;
-  border-radius: 0.2em;
+  border-radius: 2px;
   border: 0;
   outline-style: none;
   min-height: 0.4em;
   min-width: 0.4em;
-  padding: 0.2em 0;
+  padding: 2px 0;
 }
```
Comment (Nilvus) [inline data/themes/darktable.css:850]: Why changing them? Having related sizes is better here and 0.2em works great with your changes. So I suggest to put them back as that don't really change your PR.

File: data/themes/darktable.css [hunk 4]
```diff
@@ -825,16 +841,16 @@ dialog combobox window
...
 
 menu menuitem {
-  padding: 0 0.2em;
+  padding: 0 2px;
```
Comment (Nilvus) [inline data/themes/darktable.css:853]: Same here as above.

**Review by Nilvus — CHANGES_REQUESTED** [review]
Great work. Combobox are better. Thanks for that work but just some minor comments and things to change on your PR.

File: data/themes/darktable.css (hunk 2 above)
Comment (Nilvus) [reply data/themes/darktable.css:1092]: Of course it works with it too.

**Comment by Nilvus** [comment]
>     * align button to the left in preferences

Last thing, buttons as they are actually was wanted. But testing your proposal make me think that something between actual state and your proposal would be better. Like just 1px between entry and buttons and between buttons.

File: data/themes/darktable.css [hunk 5]
```diff
@@ -1148,9 +1164,13 @@ dialog scrollbar
...
 
+#preferences_notebook combobox menuitem
+{
+  padding: 0.2em;
+}
+
 #preferences_notebook button
 {
-  margin: 0 2px;
   padding: 0 2px;
```
Comment (Nilvus) [inline data/themes/darktable.css:1174]: Just made some test and I think it would be better with margin like that:
`
margin-left: 1px;`

**Review by Nilvus — CHANGES_REQUESTED** [review]

**Comment by TurboGit** [comment]
Thanks !

**Pull request merged**